### ISSUE TITLE
clang++-12 crashes when parsing this C++17 program with inline variable templates

### ISSUE THREAD

**Issue by 838895e7-073a-4e8c-821e-aa339b0dbf9d** [llvm/llvm-project#50869]
|  |  |
| --- | --- |
| Bugzilla Link | [51527](https://llvm.org/bz51527) |
| Version | 12.0 |
| OS | Linux |
| Attachments | [The code that triggers the crash](https://user-images.githubusercontent.com/93435521/143762445-ae36d460-01fb-4fb6-a541-72808542bd75.gz), [crash report shell script produced by clang](https://user-images.githubusercontent.com/93435521/143762446-80e44c01-1181-4fd1-a306-2943d83deb00.gz), [stack trace captured in markdown](https://user-images.githubusercontent.com/93435521/143762447-5b52db67-36ac-422d-9ca4-5892cc5c081e.gz) |
| CC | @AaronBallman,@DougGregor,@zygoloid,@oToToT |

## Extended Description 
clang seems to crash when parsing this C++17 code.  I've minimized the code to a point where it shouldn't successfully compile, but still crashes.

See the attachments for the source.  This is using the Debian 10 packages from apt.llvm.org.  Version details below.  Source attached.

$ clang++-12 -std=c++17 -Wall -c -o foo.o foo.cc
foo.cc:10:22: error: no template named 'Bar'
constexpr auto bar = Bar<args...>::value;
                     ^
PLEASE submit a bug report to https://bugs.llvm.org/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.      Program arguments: clang++-12 -std=c++17 -Wall -c -o foo.o foo.cc
1.      foo.cc:16:35: current parser token ';'
2.      foo.cc:13:1: parsing struct/union/class body 'Baz'
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
/usr/lib/x86_64-linux-gnu/libLLVM-12.so.1(_ZN4llvm3sys15PrintStackTraceERNS_11raw_ostreamEi+0x23)[0x7f948bbf7b23]
/usr/lib/x86_64-linux-gnu/libLLVM-12.so.1(_ZN4llvm3sys17RunSignalHandlersEv+0x50)[0x7f948bbf5e50]
/usr/lib/x86_64-linux-gnu/libLLVM-12.so.1(_ZN4llvm3sys15CleanupOnSignalEm+0xdd)[0x7f948bbf727d]
/usr/lib/x86_64-linux-gnu/libLLVM-12.so.1(+0xc4bfd6)[0x7f948bb46fd6]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x12730)[0x7f948aeec730]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang4Sema35SubstituteExplicitTemplateArgumentsEPNS_20FunctionTemplateDeclERNS_24TemplateArgumentListInfoERN4llvm15SmallVectorImplINS_23DeducedTemplateArgumentEEERNS6_INS_8QualTypeEEEPSA_RNS_4sema21TemplateDeductionInfoE+0x579)[0x7f94896d28f9]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(+0x158b363)[0x7f9489718363]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang4Sema27runWithSufficientStackSpaceENS_14SourceLocationEN4llvm12function_refIFvvEEE+0x2b)[0x7f94891b63cb]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang4Sema23DeduceTemplateArgumentsEPNS_20FunctionTemplateDeclEPNS_24TemplateArgumentListInfoENS_8QualTypeERPNS_12FunctionDeclERNS_4sema21TemplateDeductionInfoEb+0x198)[0x7f94896d6518]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang4Sema23DeduceTemplateArgumentsEPNS_20FunctionTemplateDeclEPNS_24TemplateArgumentListInfoERPNS_12FunctionDeclERNS_4sema21TemplateDeductionInfoEb+0x15)[0x7f94896da405]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang4Sema43ResolveSingleFunctionTemplateSpecializationEPNS_12OverloadExprEbPNS_14DeclAccessPairE+0x2ac)[0x7f948961139c]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(+0x154fb9e)[0x7f94896dcb9e]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang4Sema14DeduceAutoTypeENS_7TypeLocERPNS_4ExprERNS_8QualTypeEN4llvm8OptionalIjEEb+0x712)[0x7f94896dab52]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang4Sema14DeduceAutoTypeEPNS_14TypeSourceInfoERPNS_4ExprERNS_8QualTypeEN4llvm8OptionalIjEEb+0x20)[0x7f94896da430]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang4Sema28deduceVarTypeFromInitializerEPNS_7VarDeclENS_15DeclarationNameENS_8QualTypeEPNS_14TypeSourceInfoENS_11SourceRangeEbPNS_4ExprE+0x53d)[0x7f94892f931d]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang4Sema29DeduceVariableDeclarationTypeEPNS_7VarDeclEbPNS_4ExprE+0x65)[0x7f94892f9725]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang4Sema20AddInitializerToDeclEPNS_4DeclEPNS_4ExprEb+0x19a)[0x7f94892f9c9a]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6Parser30ParseCXXClassMemberDeclarationENS_15AccessSpecifierERNS_16ParsedAttributesERKNS0_18ParsedTemplateInfoEPNS_21ParsingDeclRAIIObjectE+0x13fe)[0x7f9488bc323e]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6Parser41ParseCXXClassMemberDeclarationWithPragmasERNS_15AccessSpecifierERNS0_25ParsedAttributesWithRangeENS_17TypeSpecifierTypeEPNS_4DeclE+0x735)[0x7f9488bc53c5]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6Parser27ParseCXXMemberSpecificationENS_14SourceLocationES1_RNS0_25ParsedAttributesWithRangeEjPNS_4DeclE+0x97d)[0x7f9488bbf92d]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6Parser19ParseClassSpecifierENS_3tok9TokenKindENS_14SourceLocationERNS_8DeclSpecERKNS0_18ParsedTemplateInfoENS_15AccessSpecifierEbNS0_15DeclSpecContextERNS0_25ParsedAttributesWithRangeE+0x1534)[0x7f9488bbd324]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6Parser26ParseDeclarationSpecifiersERNS_8DeclSpecERKNS0_18ParsedTemplateInfoENS_15AccessSpecifierENS0_15DeclSpecContextEPNS0_18LateParsedAttrListE+0x33f)[0x7f9488b9e43f]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6Parser35ParseSingleDeclarationAfterTemplateENS_17DeclaratorContextERKNS0_18ParsedTemplateInfoERNS_21ParsingDeclRAIIObjectERNS_14SourceLocationERNS_16ParsedAttributesENS_15AccessSpecifierE+0x385)[0x7f9488c30d35]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6Parser40ParseTemplateDeclarationOrSpecializationENS_17DeclaratorContextERNS_14SourceLocationERNS_16ParsedAttributesENS_15AccessSpecifierE+0x4ce)[0x7f9488c3011e]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6Parser36ParseDeclarationStartingWithTemplateENS_17DeclaratorContextERNS_14SourceLocationERNS_16ParsedAttributesENS_15AccessSpecifierE+0xe4)[0x7f9488c2faf4]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6Parser16ParseDeclarationENS_17DeclaratorContextERNS_14SourceLocationERNS0_25ParsedAttributesWithRangeEPS2_+0x293)[0x7f9488b9dae3]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6Parser24ParseExternalDeclarationERNS0_25ParsedAttributesWithRangeEPNS_15ParsingDeclSpecE+0xac)[0x7f9488c3e7dc]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6Parser17ParseTopLevelDeclERNS_9OpaquePtrINS_12DeclGroupRefEEEb+0x4ff)[0x7f9488c3d7ff]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang8ParseASTERNS_4SemaEbb+0x2ed)[0x7f9488b8e2ad]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang14FrontendAction7ExecuteEv+0x48)[0x7f948a13c738]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang16CompilerInstance13ExecuteActionERNS_14FrontendActionE+0x8a1)[0x7f948a0ca421]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang25ExecuteCompilerInvocationEPNS_16CompilerInstanceE+0x632)[0x7f948a19f1a2]
clang++-12(_Z8cc1_mainN4llvm8ArrayRefIPKcEES2_Pv+0x9a2)[0x413352]
clang++-12[0x4116ce]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(+0x1c4f712)[0x7f9489ddc712]
/usr/lib/x86_64-linux-gnu/libLLVM-12.so.1(_ZN4llvm20CrashRecoveryContext9RunSafelyENS_12function_refIFvvEEE+0xdd)[0x7f948bb46dad]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZNK5clang6driver10CC1Command7ExecuteEN4llvm8ArrayRefINS2_8OptionalINS2_9StringRefEEEEEPNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEPb+0x139)[0x7f9489ddbf19]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZNK5clang6driver11Compilation14ExecuteCommandERKNS0_7CommandERPS3_+0x17f)[0x7f9489db134f]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZNK5clang6driver11Compilation11ExecuteJobsERKNS0_7JobListERN4llvm15SmallVectorImplISt4pairIiPKNS0_7CommandEEEE+0x77)[0x7f9489db1707]
/usr/lib/x86_64-linux-gnu/libclang-cpp.so.12(_ZN5clang6driver6Driver18ExecuteCompilationERNS0_11CompilationERN4llvm15SmallVectorImplISt4pairIiPKNS0_7CommandEEEE+0xdc)[0x7f9489dc61bc]
clang++-12(main+0x2444)[0x410fa4]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xeb)[0x7f9487ccd09b]
clang++-12(_start+0x2a)[0x40e88a]
clang: error: clang frontend command failed with exit code 139 (use -v to see invocation)
Debian clang version 12.0.1-++20210804103842+fed41342a82f-1~exp1~20210804084624.136
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
clang: note: diagnostic msg:
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang: note: diagnostic msg: /tmp/foo-f55deb.cpp
clang: note: diagnostic msg: /tmp/foo-f55deb.sh
clang: note: diagnostic msg:

********************

**Comment by junaire** [comment]
Still reproduced in main, more simple example: https://godbolt.org/z/9PWMz4dGE

**Comment by wheatman** [comment]
This was fixed between 17.0.1 and trunk(a1358225c5b791fa9d901e74fb09606189c0e6af)
https://godbolt.org/z/Trox4c75x

code
```
struct Foo {};

template <const Foo& arg>
constexpr auto bar = Bar<arg>;

template <typename T> 
struct Baz
{
  static constexpr Foo a;
  static constexpr auto b = bar<a>;
};

```

new error
```
<source>:4:22: error: use of undeclared identifier 'Bar'
    4 | constexpr auto bar = Bar<arg>;
      |                      ^
1 error generated.
Compiler returned: 1
```

old dump
```
<source>:4:22: error: use of undeclared identifier 'Bar'
    4 | constexpr auto bar = Bar<arg>;
      |                      ^
clang++: /root/llvm-project/llvm/include/llvm/Support/Casting.h:578: decltype(auto) llvm::cast(From*) [with To = clang::FunctionTemplateDecl; From = clang::NamedDecl]: Assertion `isa<To>(Val) && "cast<Ty>() argument of incompatible type!"' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: /opt/compiler-explorer/clang-assertions-17.0.1/bin/clang++ -gdwarf-4 -g -o /app/output.s -mllvm --x86-asm-syntax=intel -S --gcc-toolchain=/opt/compiler-explorer/gcc-13.2.0 -fcolor-diagnostics -fno-crash-diagnostics -std=c++17 <source>
1.	<source>:10:35: current parser token ';'
2.	<source>:7:1: parsing struct/union/class body 'Baz'
 #0 0x00000000036e7558 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x36e7558)
 #1 0x00000000036e53dc llvm::sys::CleanupOnSignal(unsigned long) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x36e53dc)
 #2 0x0000000003632918 CrashRecoverySignalHandler(int) CrashRecoveryContext.cpp:0:0
 #3 0x00007fedd7d8d420 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x14420)
 #4 0x00007fedd785000b raise (/lib/x86_64-linux-gnu/libc.so.6+0x4300b)
 #5 0x00007fedd782f859 abort (/lib/x86_64-linux-gnu/libc.so.6+0x22859)
 #6 0x00007fedd782f729 (/lib/x86_64-linux-gnu/libc.so.6+0x22729)
 #7 0x00007fedd7840fd6 (/lib/x86_64-linux-gnu/libc.so.6+0x33fd6)
 #8 0x00000000066bb316 clang::Sema::ResolveSingleFunctionTemplateSpecialization(clang::OverloadExpr*, bool, clang::DeclAccessPair*, clang::TemplateSpecCandidateSet*) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x66bb316)
 #9 0x000000000688e01d ResolveOverloadForDeduction(clang::Sema&, clang::TemplateParameterList*, clang::Expr*, clang::QualType, bool, clang::TemplateSpecCandidateSet*) SemaTemplateDeduction.cpp:0:0
#10 0x000000000688e58e DeduceTemplateArgumentsFromCallArgument(clang::Sema&, clang::TemplateParameterList*, unsigned int, clang::QualType, clang::Expr*, clang::sema::TemplateDeductionInfo&, llvm::SmallVectorImpl<clang::DeducedTemplateArgument>&, llvm::SmallVectorImpl<clang::Sema::OriginalCallArg>&, bool, unsigned int, unsigned int, clang::TemplateSpecCandidateSet*) SemaTemplateDeduction.cpp:0:0
#11 0x000000000689eb65 clang::Sema::DeduceAutoType(clang::TypeLoc, clang::Expr*, clang::QualType&, clang::sema::TemplateDeductionInfo&, bool, bool, clang::TemplateSpecCandidateSet*) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x689eb65)
#12 0x000000000611d1cd clang::Sema::deduceVarTypeFromInitializer(clang::VarDecl*, clang::DeclarationName, clang::QualType, clang::TypeSourceInfo*, clang::SourceRange, bool, clang::Expr*) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x611d1cd)
#13 0x000000000611d9e7 clang::Sema::DeduceVariableDeclarationType(clang::VarDecl*, bool, clang::Expr*) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x611d9e7)
#14 0x000000000614dd3a clang::Sema::AddInitializerToDecl(clang::Decl*, clang::Expr*, bool) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x614dd3a)
#15 0x0000000005e45ea0 clang::Parser::ParseCXXClassMemberDeclaration(clang::AccessSpecifier, clang::ParsedAttributes&, clang::Parser::ParsedTemplateInfo const&, clang::ParsingDeclRAIIObject*) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x5e45ea0)
#16 0x0000000005e46c25 clang::Parser::ParseCXXClassMemberDeclarationWithPragmas(clang::AccessSpecifier&, clang::ParsedAttributes&, clang::TypeSpecifierType, clang::Decl*) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x5e46c25)
#17 0x0000000005e4851a clang::Parser::ParseCXXMemberSpecification(clang::SourceLocation, clang::SourceLocation, clang::ParsedAttributes&, unsigned int, clang::Decl*) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x5e4851a)
#18 0x0000000005e4a510 clang::Parser::ParseClassSpecifier(clang::tok::TokenKind, clang::SourceLocation, clang::DeclSpec&, clang::Parser::ParsedTemplateInfo const&, clang::AccessSpecifier, bool, clang::Parser::DeclSpecContext, clang::ParsedAttributes&) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x5e4a510)
#19 0x0000000005e23d78 clang::Parser::ParseDeclarationSpecifiers(clang::DeclSpec&, clang::Parser::ParsedTemplateInfo const&, clang::AccessSpecifier, clang::Parser::DeclSpecContext, clang::Parser::LateParsedAttrList*, clang::ImplicitTypenameContext) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x5e23d78)
#20 0x0000000005ee6092 clang::Parser::ParseSingleDeclarationAfterTemplate(clang::DeclaratorContext, clang::Parser::ParsedTemplateInfo const&, clang::ParsingDeclRAIIObject&, clang::SourceLocation&, clang::ParsedAttributes&, clang::AccessSpecifier) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x5ee6092)
#21 0x0000000005eec631 clang::Parser::ParseTemplateDeclarationOrSpecialization(clang::DeclaratorContext, clang::SourceLocation&, clang::ParsedAttributes&, clang::AccessSpecifier) (.part.0) ParseTemplate.cpp:0:0
#22 0x0000000005eec8a3 clang::Parser::ParseDeclarationStartingWithTemplate(clang::DeclaratorContext, clang::SourceLocation&, clang::ParsedAttributes&, clang::AccessSpecifier) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x5eec8a3)
#23 0x0000000005e2a7f9 clang::Parser::ParseDeclaration(clang::DeclaratorContext, clang::SourceLocation&, clang::ParsedAttributes&, clang::ParsedAttributes&, clang::SourceLocation*) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x5e2a7f9)
#24 0x0000000005df6584 clang::Parser::ParseExternalDeclaration(clang::ParsedAttributes&, clang::ParsedAttributes&, clang::ParsingDeclSpec*) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x5df6584)
#25 0x0000000005df7452 clang::Parser::ParseTopLevelDecl(clang::OpaquePtr<clang::DeclGroupRef>&, clang::Sema::ModuleImportState&) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x5df7452)
#26 0x0000000005deb76a clang::ParseAST(clang::Sema&, bool, bool) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x5deb76a)
#27 0x0000000004939258 clang::CodeGenAction::ExecuteAction() (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x4939258)
#28 0x00000000041a1c09 clang::FrontendAction::Execute() (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x41a1c09)
#29 0x00000000041278ce clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x41278ce)
#30 0x0000000004281de6 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x4281de6)
#31 0x0000000000bc67eb cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0xbc67eb)
#32 0x0000000000bbefaa ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&, llvm::ToolContext const&) driver.cpp:0:0
#33 0x0000000003f83419 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<std::optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const::'lambda'()>(long) Job.cpp:0:0
#34 0x0000000003632dc4 llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x3632dc4)
#35 0x0000000003f83a0f clang::driver::CC1Command::Execute(llvm::ArrayRef<std::optional<llvm::StringRef>>, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char>>*, bool*) const (.part.0) Job.cpp:0:0
#36 0x0000000003f4b275 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&, bool) const (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x3f4b275)
#37 0x0000000003f4bcdd clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&, bool) const (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x3f4bcdd)
#38 0x0000000003f5385d clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*>>&) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0x3f5385d)
#39 0x0000000000bc4a87 clang_main(int, char**, llvm::ToolContext const&) (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0xbc4a87)
#40 0x0000000000abef61 main (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0xabef61)
#41 0x00007fedd7831083 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24083)
#42 0x0000000000bbea8e _start (/opt/compiler-explorer/clang-assertions-17.0.1/bin/clang+++0xbbea8e)
clang++: error: clang frontend command failed with exit code 134 (use -v to see invocation)
Compiler returned: 134
```
